sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/GenericTypeParameterUnused.cs

```diff
@@ -57,18 +57,19 @@ protected override void Initialize(SonarAnalysisContext context)
                         }
 
                         var helper = CreateParametersInfo(c.Node, c.SemanticModel);
-                        if (helper?.Parameters == null || helper.Parameters.Parameters.Count == 0)
+                        if (helper.Parameters == null || helper.Parameters.Parameters.Count == 0)
                         {
                             return;
                         }
 
                         var declarations = declarationSymbol.DeclaringSyntaxReferences
                             .Select(reference => reference.GetSyntax());
 
-                        var usedTypeParameters = GetUsedTypeParameters(declarations, c, analysisContext.Compilation);
+                        var typeParameterNames = helper.Parameters.Parameters.Select(typeParameter => typeParameter.Identifier.Text).ToList();
```

Comment:
maybe make it a set (see below comment)?

Reply:
It depends on the epected size: https://stackoverflow.com/questions/150750/hashset-vs-list-performance
If a set is not the best way to go (collection to small), consider an array as it read-only (right?).

Reply:
Looking at the graph (https://stackoverflow.com/a/10762995), in the vast majority of cases we should be fine with a list since usually the methods have just a few type parameters.

I've replaced the list with an array but I expect that the optimizations done at this level would be rather small.

Reply:
ok, basically my comment was not appropiate at small hash sizes given  that the list is actually used for type parameters

my bad :)

Reply:
array v.s. list would indeed make no (real) difference, but I think you should minimize the overhead if you just need an internal read-only collection. :)

Reply:
@andrei-epure-sonarsource: I think the idea is very good and that it's important to keep this in mind and make conscious decisions when picking collections so I'm very glad you asked (it's so easy to forget and select the list by default)

@Corniel: indeed, and this will make a difference in the end if used consistently :)